### ISSUE TITLE
Easy navigate to page level sections (Toolbar)(New component)

### ISSUE THREAD

**Issue by kirkian** [vmware/clarity#3777]
```
[ ] bug
[x] feature request
[ ] enhancement
```

## Summary 
Keyboard-power and accessibility users should have a way to reach any section within the page. As the most common techniques in web accessibility is the “Skip Navigation” link, sometimes referred to as “Skip to Content,” however, in enterprise products with lots of content and many ways to navigate to the sections with a page, there is a need to have a toolbar with shortcuts to the main sections.

## Description
The objective of this toolbar is to provide a mechanism to bypass the sections of the page that are repeated on multiple web pages by skipping directly to any part within a page. This also covers accessibility success Criterion [2.4.1 Bypass Blocks](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html)

## Toolbar
1. **Navigate to**: contains each page main sections
2. **Press OPT/ALT + LETTER** is the keyboard shortcut which opens the toolbar

![a11y-toolbar](https://user-images.githubusercontent.com/2523188/64172837-e8d88200-ce66-11e9-828d-5878566f6b4d.png)

**Color Contrast**: It passes AAA level

### Interaction when the page loads
- Hit tab opens a toolbar
- By default highlighting sections in this page dropdown (Select>Option)
- Using up/down arrow keys while highlighting the option in the dropdown the section in the page will be focused
- When choosing an option from the dropdown, the selected option will be focused on the page

### Interaction using a shortcut
- Hit a predefined shortcut, e.g.: opt + t
- By default highlighting sections in this page dropdown (Select>Option)
- Using up/down arrow keys while highlighting the option in the dropdown the section in the page will be focused
- When choosing an option from the dropdown, the selected option will be focused on the page

* The toolbar will be hidden when the user selects an option or hit tab again

### Page layout in default mode 
![Home](https://user-images.githubusercontent.com/2523188/64121730-bf5d1f00-cdb0-11e9-9a11-6c57bc360cf3.png)

### Page layout showing A11y toolbar
The toolbar is visible when the user hits tab after page loads or on-demand using a shortcut
![Home_Toolbar](https://user-images.githubusercontent.com/2523188/64172865-f2fa8080-ce66-11e9-9f57-00080e996f76.png)

### Navigate to section with the section highlight
![Home_Performance](https://user-images.githubusercontent.com/2523188/64172873-f68e0780-ce66-11e9-95d4-62c5a8ac1387.png)

**Comment by kirkian** [comment]
@Jinnie Hi Ivan, I think this should go under `New Component` label.

**Comment by gnomeontherun** [comment]
@chlane This is a request for a more expansive keyboard only toolbar to appear (like the skip link but with a document outline). Do you have any input or suggestions on this?

**Comment by chlane** [comment]
This appears to be a `<select>` element allowing the user to set focus to different parts of the pages which will work.  But the  opt + t shortcut doesn't provide any accessibility benefit since the user should just be able to Tab to the '<select>' and press Enter to activate it.

**Comment by gnomeontherun** [comment]
Couple of notes

- Dark theme needs consideration
- Toolbar is pushing down, but that is a problem with layouts, need to understand options here
- The select/dropdown is not a component, should probably just use a menu or select instead (which already have keyboard shortcuts)
- Need to review other examples already out there (Starbucks comes to mind)

**Comment by kirkian** [comment]
Thanks for the responses, guys!

- Dark theme the toolbar colors are the same, except the dropdown
- When the toolbar bar appears the options in the list changes based on the page section
- The select/dropdown is the dropdown component https://v2.clarity.design/dropdowns
- I already did study research and will do a review with my team for this toolbar.

cc @sheribyrne as I demoed about the toolbar couple days ago.

Let me know if you need clarification.

Thanks
Nareg

**Comment by gnomeontherun** [comment]
The feature request here has been captured into our list and we’re going to take it into consideration as we develop Clarity Core capabilities. In an effort to clean up our backlog and focus our attention, I’m going to close this as captured in our feature requests. Please follow our development and releases to see when we release relevant components to make this possible. Future feature requests can be made in our [GitHub Discussions](https://github.com/vmware/clarity/discussions).
